### ISSUE TITLE
Glide.clear seems to work in non-expected way

### ISSUE THREAD

**Issue by alexzaitsev** [bumptech/glide#1601]
Hello. 
I'm not sure that the title of this issue is good enough so please keep reading. 
I have a gifs inside custom adapter, inherited from BaseAdapter. 
Let me describe the flow. 

In `getView` I create an url order which is based on the business logic. Basically it contains several gif urls. Some of the can be already cached by Glide. I try to load first url. If it fails (url is wrong or network connection was broken) - I try to get next url and load it. 
```
                    @Override
                    public boolean onException(Exception e, String model, Target<GlideDrawable> target, boolean isFirstResource) {
                        proceedLoadQueue();
                        return false;
                    }
```
Finally, when the order is empty (it means that no gif was loaded into ImageView) I show `Retry` button. It's in front of ImageView:
```
            retryBtn.setVisibility(View.VISIBLE);
            progress.setVisibility(View.GONE);
            retryBtn.setTag(true);
            target.setOnClickListener(null);
```
And further in `getView` if retryBtn contains tag I don't even try to load my queue, I just show `Retry` button. It works almost as expected except one thing. If I will scroll my list fast up and down, after several frictions some gifs can appear on the neighbour ImageView as well. Or it could be replaced by `Retry` button. I checked these stuff very well and I'm pretty sure that it's some non-expected behaviour of Glide. So I tried to add `Glide.clear(imageView);` before gif loading. But if I add this my adapter shows incorrect gifs even after single up-down scroll. (ImageView doesn't have `setTag` code).

Long story short: I suggest that Glide don't clear ImageView before loading or something similar that's why I can see some gifs on the wrong places. 
I think that I can explain why Retry button appears after some scrolls: Glide fails to load a gif (I don't know why, but let it be) and that's why Retry button is shown. It's ok behaviour.

How can I be 100% sure that I'm loading a gif into clear ImageView?

**Comment by TWiStErRob** [comment]
You don't need to call `Glide.clear` before, because `into()` clears if necessary. However if you have complex branching logic you need to be sure that all possible branching paths clear the target. The simplest example is:
```java
if(model.hasImage()) {
    Glide...load(model).into(holder.view);
} else {
    Glide.clear(holder.view);
}
```
without the else branch stale images may stale on positions where there shouldn't be any.

Regarding the Retry button it may be something similar. Whether the retry button is shown for a certain position should be stored in your model inside the adapter, otherwise information (tag) may leak from one position to another when recycling happens. Remember that upon binding another position to the item, the view of that item stays untouched (as left by the previous bind). So in order to have consistent behavior you either need to reset your view at the beginning of bind, or use the appropriate callback to reset the view when it goes out of screen. e.g. use [`RecyclerView.Adapter.onViewRecycled`](https://developer.android.com/reference/android/support/v7/widget/RecyclerView.Adapter.html#onViewRecycled(VH)) to reset the view state to the same that was just after `onCreateViewHolder` so further binds can set it up from a clean state.

**Comment by alexzaitsev** [comment]
Ok, I changed my approach and it works now. Thanks.
